### ISSUE TITLE
Encrypt chat history in database with the profile key

### ISSUE THREAD

**Issue by roihershberg** [evilcorpltd/aTox#920]

**Comment by robinlinden** [comment]
No work should be done on this yet unless we can deal with it separately from the database library used. The library we're using for db management (Room) is being replaced to enable multi-platform.

**Comment by roihershberg** [comment]
What about [this](https://sonique6784.medium.com/protect-your-room-database-with-sqlcipher-on-android-78e0681be687)?
It seems very simple and easy

**Comment by robinlinden** [comment]
Yeah, integrating it into Room is quick and easy, but we'll have to set up migrations from Room to the other (I've been testing SQLDelight) without losing any data, so I didn't want to involve encryption before that migration is done to make things easy.

**Comment by roihershberg** [comment]
Yeah you're right

**Comment by robinlinden** [comment]
We're sticking with Room. They ended up adding support for Kotlin multiplatform before we switched, so this isn't blocked by a migration to SQLDelight anymore.
